src/main/java/com/ldtteam/blockui/views/ScrollingListContainer.java

```diff
@@ -3,36 +3,82 @@
+import com.ldtteam.blockui.util.SafeError;
+import com.ldtteam.blockui.util.records.SizeI;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
 
 /**
  * A Blockout pane that contains a scrolling line of other panes.
  */
 public class ScrollingListContainer extends ScrollingContainer
 {
-    private int listElementHeight = 0;
+    /**
+     * The xml parameters for the row panes.
+     */
+    @Nullable
+    private PaneParams listNodeParams;
+
+    /**
+     * The template size for the rows.
+     */
+    private SizeI templateSize = new SizeI(0, 0);
 
     ScrollingListContainer(final ScrollingList owner)
     {
         super(owner);
     }
 
+    /**
+     * Set the new xml parameters for the row panes.
+     *
+     * @param listNodeParams the new xml parameters.
+     */
+    public void setListNodeParams(final @NotNull PaneParams listNodeParams)
+    {
+        this.listNodeParams = listNodeParams;
+
+        final Pane template = Loader.createFromPaneParams(listNodeParams, null);
+        if (template == null)
+        {
+            SafeError.throwOrLog(new IllegalStateException("Scrolling list template could not be loaded. Is there a reference to another layout in the list children?"));
+            return;
+        }
+
+        this.templateSize = new SizeI(template.getWidth(), template.getHeight());
+    }
+
     /**
      * Creates, deletes, and updates existing Panes for elements in the list based on the DataProvider.
      *
-     * @param dataProvider   data provider object, shouldn't be null.
-     * @param listNodeParams the xml parameters for this pane.
+     * @param dataProvider data provider object, shouldn't be null.
+     * @param height       the maximum height of the parent.
+     * @param childSpacing the spacing between each row.
      */
-    public void refreshElementPanes(final ScrollingList.DataProvider dataProvider, final PaneParams listNodeParams, final int height, final int childSpacing)
+    public void refreshElementPanes(final ScrollingList.DataProvider dataProvider, final int height, final int childSpacing)
     {
+        int currentYpos = 0;
+
+        if (listNodeParams == null)
+        {
+            SafeError.throwOrLog(new IllegalStateException("Template size is not defined. Does the scrolling list have a child?"));
+            return;
+        }
+
         final int numElements = (dataProvider != null) ? dataProvider.getElementCount() : 0;
-        if (dataProvider != null)
+        if (numElements > 0)
         {
+            final RowSizeModifier modifier = new RowSizeModifier();
+
             for (int i = 0; i < numElements; ++i)
             {
-                final Pane child;
-                final int childYpos = (childSpacing + listElementHeight) * i;
-                if (childYpos + listElementHeight >= scrollY && childYpos <= scrollY + height)
+                modifier.reset(templateSize.width(), templateSize.height());
```

Comment:
why do we reset it and then reapply it?

Reply:
It's single instance, similar logic to blockpos streams - to avoid creating object for each list element

Reply:
Would it make sense to either have the reset within the application itself?

Reply:
It's reset for next use and prepare method, it's not even public.. (read the code it's almost one liner)
This entire class exists since you can't return two values and since we don't want to call set size if the size hasn't changed